### DISCUSSION TITLE
Add option to set units according to variables.

### DISCUSSION THREAD

**Discussion by erturkmemmedli** [grafana/grafana#46514] [Dashboards, Panels & Visualizations]
I am using variables to dynamically change my dashboard.

There are three variables, where two of them represents gas fields and sub-fields accordingly. When you choose a field in "fields" variable, it automatically gives you option to choose related sub-fields under "sub-fields" variables. The problematic variable is the third one, which determines gas parameters, such as temperature, pressure, etc.

But one dashboard does not allow us to set units of parameters according to variables.

If there was an option to set each unit depending on variables, it would be a great feature.

It can also apply other settings, such as title of the dashboard.

Assume you select temperature under "parameters" variable, dashboard is automatically named as "Temperature of Something", and the values shown in the dashboard are automatically followed by "°C" unit.

**Comment by torkelo** [comment]
Not sure this makes any sense, variables are for filtering data, not display settings. 

You can use the transforms config from query or rows to fields to extract display settings from your query result (and apply it to other parts of the query results)

**Reply by ghost** [reply]
Maybe it does make sense if you have a query that returns several series in different units, and you don't want to punch all series on the same Y axis. Now, if the label set for each series already contains the unit, it would be great if I could just put ${__field.labels.unit} into the Unit field, as is possible with the Display Name.
Prometheus people who suffix their metric names with _someunit will probably also like this.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
